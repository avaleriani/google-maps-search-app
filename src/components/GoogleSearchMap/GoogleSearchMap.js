import React from "react";
import Geocode from "react-geocode";
import MapComponent from "../MapContainer/MapContainer";
import Button from "../Button/Button";
import styles from "./styles.module.css";


class GoogleSearchComponent extends React.Component {
  constructor() {
    super();
    Geocode.setApiKey(process.env.REACT_APP_GOOGLE_MAPS_API_KEY);
    this.state = {
      lat: 0,
      lng: 0,
      location: null
    };
    this.search = this.search.bind(this);
    this.onInput = this.onInput.bind(this);
  }

  search() {
    Geocode.fromAddress(this.state.location).then(
      response => {
        const { lat, lng } = response.results[0].geometry.location;
        this.setState({
          lat: lat,
          lng: lng
        })
      },
      error => {
        console.error(error);
      }
    );
  }

  onInput(evt) {
    this.setState({
      location: evt.target.value
    })
  }

  render() {
    return (
      <div className={styles["google-search-map"]}>
        <MapComponent lat={this.state.lat} lng={this.state.lng}/>
        <input
          type="text"
          placeholder="Where do you want to go?"
          className={styles["search-box"]}
          onChange={this.onInput}
        />
        <Button className={styles["button"]} onClick={this.search} label={"Search"}/>

      </div>
    )
  };
}

export default GoogleSearchComponent;
