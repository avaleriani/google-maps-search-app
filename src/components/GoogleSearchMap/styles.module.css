.google-search-map{
    position: relative;
    font-size: 14px;
    padding: 20px;
}
.search-box {
    position: relative;
    width: 200px;
    height: 32px;
    top: 0;
    padding: 0 12px;
    border-radius: 3px;
    box-shadow: 0 2px 6px rgba(0, 0, 0, 0.3);
    outline: none;
}

.button {
    position: relative;
    right: 0;
    width: 72px;
    height: 38px;
    top: 0;
    outline: none;
}
