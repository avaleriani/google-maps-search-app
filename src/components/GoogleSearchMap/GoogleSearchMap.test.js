import React from 'react';
import GoogleSearchMap from './GoogleSearchMap';
import renderer from 'react-test-renderer';


it('renders correctly', () => {
  const component = renderer.create(
    <GoogleSearchMap/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});