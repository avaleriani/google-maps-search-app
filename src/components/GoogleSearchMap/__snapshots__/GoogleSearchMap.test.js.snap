// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="google-search-map"
>
  <div>
    Loading Map...
  </div>
  <input
    className="search-box"
    onChange={[Function]}
    placeholder="Where do you want to go?"
    type="text"
  />
  <button
    className="button"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
